### PR TITLE
Bugfixes for the s390x and ppc architecture

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -242,6 +242,9 @@ if test -z "${USING_WIN32_FALSE}" && test "${have_msg_no_signal}" = "no" && \
  AC_MSG_ERROR([Your system needs either MSG_NOSIGNAL or SO_NOSIGPIPE])
 fi
 
+AC_CHECK_TYPE(termios2,
```
Comment (peternewman) [inline configure.ac]: Will you not need to include asm/termios.h here to find it?

File: plugins/usbdmx/LibUsbThreadTest.cpp [hunk 2]
```diff
@@ -87,6 +87,10 @@ void LibUsbThreadTest::testNonHotplug() {
 
 #if defined(LIBUSB_API_VERSION) && (LIBUSB_API_VERSION >= 0x01000102)
 void LibUsbThreadTest::testHotplug() {
+  if (!m_context) {
+    return;
```
Comment (peternewman) [inline plugins/usbdmx/LibUsbThreadTest.cpp]: Can we have a message here so we know its not running all the tests.

**Comment by peternewman** [comment]
LGTM aside from comments.

File: configure.ac (hunk 1 above)
Comment (nomis52) [inline configure.ac]: Yup, added :)

File: plugins/usbdmx/LibUsbThreadTest.cpp (hunk 2 above)
Comment (nomis52) [inline plugins/usbdmx/LibUsbThreadTest.cpp]: There are other reasons for this not running, like the #if checks. I don't want to get into the pattern of adding logs for all of these.

Comment (peternewman) [inline plugins/usbdmx/LibUsbThreadTest.cpp]: But there's two classes of skipping here, the #if, which means the code in olad won't run either and this check, which means the olad code should run if it's got access to /sys or whatever is blocked here. I just think we should warn where we've got a coverage gap and the tests aren't fully testing the code you're able to run in olad on that machine.

Comment (nomis52) [inline plugins/usbdmx/LibUsbThreadTest.cpp]: We already have the log line on 68 that lets us infer this.

Comment (peternewman) [inline plugins/usbdmx/LibUsbThreadTest.cpp]: Ah yes, fair point.

**Pull request merged**